workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -354,15 +385,15 @@ private void SetupDispatcherHandlers()
                 .SelectMany(assembly => assembly.GetTypes())
                 .Where(type => typeof(ComponentDispatcherHandler).IsAssignableFrom(type) && !type.IsAbstract);
 
+            worker.TryGetEntity(new EntityId(Worker.WorkerEntityId), out var workerEntity);
```

Comment:
Wait why are we giving the WorkerEntity a `SpatialEntityId` ?

Reply:
because it's looking it up in the entitymapping :P I got rid of all the special cases. The other option I thought about would be to have a direct reference to the WorkerEntity inside the Worker

Reply:
I think that'd be preferable - imagine the case:

```
var cg = CreateComponentGroup(typeof(SpatialEntityId), typeof(RandomCommandSender);
```

You now have injected the WorkerEntity as well